source/h1_connection.c

```diff
@@ -721,6 +731,82 @@ static void s_client_update_incoming_stream_ptr(struct aws_h1_connection *connec
     s_set_incoming_stream_ptr(connection, desired);
 }
 
+static void s_http_stream_idle_timeout_task(struct aws_task *task, void *arg, enum aws_task_status status) {
+    (void)task;
+    struct aws_h1_stream *stream = arg;
+    struct aws_http_connection *connection_base = stream->base.owning_connection;
+    if (status == AWS_TASK_STATUS_CANCELED) {
+        AWS_ZERO_STRUCT(stream->base.client_data->idle_timeout_task);
+        return;
+    }
+
+    struct aws_h1_connection *connection = AWS_CONTAINER_OF(connection_base, struct aws_h1_connection, base);
+    /* Timeout happened, close the connection */
+    uint64_t idle_timeout_ms = stream->base.client_data->idle_timeout_ms == 0
+                                   ? connection_base->client_data->idle_timeout_ms
+                                   : stream->base.client_data->idle_timeout_ms;
+    AWS_LOGF_INFO(
+        AWS_LS_HTTP_CONNECTION,
+        "id=%p: Closing connection as timeout after request sent to the first byte received happened. Idle time "
+        "is %" PRIu64 " ms",
+        (void *)connection_base,
+        idle_timeout_ms);
+
+    AWS_ZERO_STRUCT(stream->base.client_data->idle_timeout_task);
```

Comment:
trivial: maybe just zero this at the top of the function, instead of doing it from 2 different places

/* zero-out task to indicate that it's no longer scheduled */